hazelcast/src/test/java/com/hazelcast/internal/nearcache/AbstractNearCacheSerializationCountTest.java

```diff
@@ -200,13 +219,19 @@ private void runTest(NearCacheTestContext<KeySerializationCountingData, ValueSer
                 }
                 assertAndResetSerializationCounts(context, "put()", 0);
 
-                resultValue = context.nearCacheAdapter.get(key);
+                if (!isCacheOnUpdate(nearCacheConfig) && isInvalidateOnChange(nearCacheConfig)) {
```

Comment:
We do not spawn a new instance if invalidations are disabled, to speed up the test. We also cannot use a new instance if `LocalUpdatePolicy.CACHE_ON_UPDATE` is configured, since we need the populated Near Cache to read from. In all other cases we spawn a new Near Cache instance here, so we don't have a race with the invalidation events.